src/megaclient.cpp

```diff
@@ -4270,6 +4336,15 @@ bool MegaClient::procsc()
                             fetchingnodes = false;
                             restag = fetchnodestag;
                             fetchnodestag = 0;
+
+                            using CType = CacheableStatus::Type;
+                            auto cachedBlockedState = mCachedStatus[CType::STATUS_BLOCKED];
+                            if (!mBlockedSet && cachedBlockedState && cachedBlockedState->value()) //block state not received in this execution, and cached says we were blocked last time
+                            {
+                                LOG_debug << "cached blocked states reports blocked, and no block state has been received before, issuing whyamiblocked";
+                                whyamiblocked();// lets query again, to trigger transition and restoreSyncs
```

Comment:
Now that the SDK may have the blocked state in cache and triggers the `whyamiblocked()` in case the current state is still unknown, maybe we should ask QA to test, for this ticket (Risk Areas), one of the cases where the mobile apps have seen a blocked status (so added into cache) but, upon restart the app, the status is unblocked. Just to confirm it doesn't introduce any buggy side-effect.

Reply:
added a note in redmine ticket. (could use some rewording ;-) )